### PR TITLE
Force Light Mode permanently on

### LINKED ISSUE #296: UI breaks if Windows switches dark/light themes
### Version

v2.4.12

### Platform

Windows 10

### Steps to reproduce

Step 1: Open LightBulb
Step 2: Switch the Windows theme in Settings

### Details

- Expected behavior: The app either does not change, or changes to the requested theme.
- Actual behavior: The app changes to the requested theme, but all the styling breaks:

image
image

The only way out is restarting the app.

The reason for this is that, when the system theme changes, MaterialTheme reloads the theme and applies the requested theme. However, it doesn't re-apply the theme created in App.axml.cs, instead using the default empty theme. Since no colors are added by default, nothing displays!

I have a fix for this, but there are several different options, and I want to get your opinion. [Here's a branch](https://github.com/LilithSilver/LightBulb/tree/dark-mode) with the changes for option 2. It still needs some work, but I wanted to get your thoughts before I go ahead on opening a PR.

### Option 1: Allow dark mode and don't touch the accent colors

Dark mode is available by default with the correct approach, and everything works OK (with an update to Material 3.6):

image
image

Light mode would remain identical (Ignore the darkened lightbulb icon; I'd fix that if we took this approach). 

However, notice how the sliders are difficult to see.

### Option 2: Allow dark mode and choose a reasonable accent color that works with light/dark

To fix the issue with option 1, a different accent color can be chosen (this is just an example):

image

Light mode would look like:
image

### Option 3: Force light mode

Light mode can be forced by setting BaseTheme="Light" on the MaterialTheme element and specifying the custom accent colors as brush resource overrides. This is the simplest option, but I'd prefer to have dark mode because light mode hurts my eyes. (And this application seems important to make easy on the eyes).

### Option 4: Change color configurations between light and dark mode

I don't know how to do this option with the Material library for Avalonia. I would likely need to open an issue and ask the dev. If changing the color configurations between dark/light mode isn't possible in the current state, it *should* be, considering the [material guidelines](https://m2.material.io/design/color/dark-theme.html) recommend it.

### Checklist

- [X] I have looked through existing issues to make sure that this bug has not been reported before
- [X] I have provided a descriptive title for this issue
- [X] I have made sure that this bug is reproducible on the latest version of the application
- [X] I have provided all the information needed to reproduce this bug as efficiently as possible
- [ ] I have sponsored this project

### PR DESCRIPTION
**Pull request by LilithSilver**
Closes #296 by forcing light mode on via the theme, and specifying theme colors in `Application.Resources` instead of by manually building a theme.

### CODE REVIEW

File: LightBulb/App.axaml [hunk 1]
```diff
@@ -11,8 +11,27 @@
...
+		<SolidColorBrush x:Key="MaterialSecondaryMidBrush" Color="#F9A825"></SolidColorBrush>
+		<SolidColorBrush x:Key="MaterialSecondaryMidForegroundBrush" Color="#F9A825"></SolidColorBrush>
+		<SolidColorBrush x:Key="MaterialSecondaryDarkBrush" Color="#F9A825"></SolidColorBrush>
+		<SolidColorBrush x:Key="MaterialSecondaryDarkForegroundBrush" Color="#F9A825"></SolidColorBrush>
+	</Application.Resources>
+	
     <Application.Styles>
-        <materialStyles:MaterialTheme />
+		<!-- Dummy values for foreground colors to initialize properly. All associated brushes are overridden in Application.Resources. -->
+        <materialStyles:MaterialTheme BaseTheme="Light" PrimaryColor="Purple" SecondaryColor="Green"/>
```
Comment (Tyrrrz) [inline LightBulb/App.axaml:34]: Are the dummy values actually necessary here? If so, what do they achieve?

File: LightBulb/App.axaml [hunk 2]
```diff
@@ -11,8 +11,27 @@
...
+		<SolidColorBrush x:Key="MaterialPrimaryDarkBrush" Color="#343838"></SolidColorBrush>
+		<SolidColorBrush x:Key="MaterialPrimaryForegroundBrush" Color="#343838"></SolidColorBrush>
+
+		<!-- Override theme Secondary color -->
+		<SolidColorBrush x:Key="MaterialSecondaryLightBrush" Color="#F9A825"></SolidColorBrush>
+		<SolidColorBrush x:Key="MaterialSecondaryLightForegroundBrush" Color="#F9A825"></SolidColorBrush>
+		<SolidColorBrush x:Key="MaterialSecondaryMidBrush" Color="#F9A825"></SolidColorBrush>
+		<SolidColorBrush x:Key="MaterialSecondaryMidForegroundBrush" Color="#F9A825"></SolidColorBrush>
+		<SolidColorBrush x:Key="MaterialSecondaryDarkBrush" Color="#F9A825"></SolidColorBrush>
+		<SolidColorBrush x:Key="MaterialSecondaryDarkForegroundBrush" Color="#F9A825"></SolidColorBrush>
```
Comment (Tyrrrz) [inline LightBulb/App.axaml:29]: I may be wrong, but shouldn't all these colors not be the same?

**Comment by Tyrrrz** [comment]
I think a simpler fix would be to set `RequestedThemeVariant="Light"` on the `<Application>` element, and leave the rest of the code as is

**Comment by Tyrrrz** [comment]
Ok fair enough. I'm just wary of setting all color brushes by hand like that, for two reasons:

1. My understanding is that, when using the current approach, the tertiary colors (such as `MaterialSecondaryDarkBrush`) are automatically derived from primary/secondary colors by way of adjusting HSL. Correct me if I'm wrong, I haven't checked how it works since ~2017.
2. If the Material theming library adds new colors in a new version, we'd have to update the resources to support them. Which could lead to some unexpected issues.

It looks like the doc you linked doesn't necessarily specify why the current approach is incorrect though, at least I don't see where it does that.

**Comment by LilithSilver** [comment]
Those points are good ones -- I played around with it a bit more and got `MaterialThemeBase` to work with an initial override that later gets set by the theme for the missing brush. (Before I was doing it via code, and it was ugly, but this axaml version looks OK to me).

Let me know what you think.

**Comment by Tyrrrz** [comment]
Looks good. Can you also please explain why `RequestedThemeVariant="Light"` doesn't work, regardless of `MaterialTheme`/`MaterialThemeBase` approach? My understanding is that it would hard-code the theme so even if the OS-wide requested theme changes, it won't affect the app.

**Comment by LilithSilver** [comment]
> Looks good. Can you also please explain why RequestedThemeVariant="Light" doesn't work, regardless of MaterialTheme/MaterialThemeBase approach? My understanding is that it would hard-code the theme so even if the OS-wide requested theme changes, it won't affect the app.

It does work -- I'll make that change as well. As long as we're correctly using MaterialThemeBase I've got no issues with it.

File: LightBulb/App.axaml [hunk 3]
```diff
@@ -6,13 +6,23 @@
...
     xmlns:materialAssists="clr-namespace:Material.Styles.Assists;assembly=Material.Styles"
     xmlns:materialIcons="clr-namespace:Material.Icons.Avalonia;assembly=Material.Icons.Avalonia"
-    xmlns:materialStyles="clr-namespace:Material.Styles.Themes;assembly=Material.Styles">
+    xmlns:materialStyles="clr-namespace:Material.Styles.Themes;assembly=Material.Styles"
+    RequestedThemeVariant="Light">
     <Application.DataTemplates>
         <framework:ViewManager />
     </Application.DataTemplates>
-
+
```
Comment (Tyrrrz) [inline LightBulb/App.axaml:14]:
```suggestion

```

**Comment by Tyrrrz** [comment]
Thanks!

**Pull request merged**